src/app/core/services/authentication.service.ts

```diff
@@ -0,0 +1,126 @@
+import { Injectable } from '@angular/core';
+import { AngularFireAuth } from '@angular/fire/compat/auth';
+import {
+  BehaviorSubject,
+  catchError,
+  from,
+  map,
+  Observable,
+  tap,
+  throwError,
+} from 'rxjs';
+import firebase from 'firebase/compat';
+import FirebaseError = firebase.FirebaseError;
+import { UsersService } from './users.service';
+import { Store } from '@ngrx/store';
+import { removeToken, setToken } from '../states/auth/auth.actions';
+import { resolveError, setErrorMessage } from '../states/error/error.actions';
+import UserCredential = firebase.auth.UserCredential;
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthenticationService {
+  loggedIn = new BehaviorSubject<boolean>(false);
+  loggedIn$ = this.loggedIn.asObservable();
+  constructor(
+    private auth: AngularFireAuth,
+    private usersService: UsersService,
+    private store: Store,
+  ) {
+    this.auth.onAuthStateChanged((user) => {
+      if (user) {
+        this.loggedIn.next(true);
+      } else {
+        this.loggedIn.next(false);
+      }
+    });
+  }
+
+  signInWithEmailPassword(
```

Comment:
The signature already tells the story so you could use `signIn` to keep the language short.
If you want to see it in your code then I would prefer using an object to expose the parameter names on the consumer side. That is probably what you liked about the long name. 
```
signIn({
  email: ...,
  password: ...
})
```

Passing an object is a common way of establishing named parameters in javascript while there is no official support for that. I like doing that.